pydantic/main.py

```diff
@@ -703,7 +703,14 @@ def __getattr__(self, item: str) -> Any:
                     except KeyError as exc:
                         raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}') from exc
                 else:
-                    raise AttributeError(f'{type(self).__name__!r} object has no attribute {item!r}')
+                    if hasattr(self.__class__, item):
+                        raise AttributeError(
```

Comment:
```suggestion
                        return super().__getattr__(self.__class__, item)
```

Don't know if this will work but I think it should be possible to raise the normal error. I'll play with it a bit